engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/TypeOfNode.java

```diff
@@ -110,7 +116,15 @@ Object doPolyglotNumber(
     if (interop.fitsInInt(proxy)) {
       return builtins.number().getInteger();
     } else if (interop.fitsInDouble(proxy)) {
-      return builtins.number().getDecimal();
+      try {
+        if (Double.isNaN(interop.asDouble(proxy))) {
+          return builtins.number().getNaN();
```

Comment:
The type is `Nan`, but here we have `getNaN`. Is `NaN` not an allowed type name in Enso?

A bit more consistency is desirable. Probably rename to `getNan`...

Reply:
Alternative is something longer like `Not_A_Number`, but that's not question for me. It is up to library guys to choose what they want.